### PR TITLE
🧪 Add tests for invites route edge cases and error paths

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** Added tests to cover the remaining untested branches in `invites.ts` related to the `PUT /api/invites/:id` endpoint (accepting/declining an invite).
📊 **Coverage:** Added specific tests for:
- Invalid action parameter resulting in 400.
- Invite not found resulting in 404.
- User is not the correct invitee resulting in 403.
- Invite already responded to resulting in 400.
- General database error propagation when accepting an invite (testing the `db.batch` failure).
✨ **Result:** `invites.ts` test coverage is now 100% and these edge cases are safeguarded.

---
*PR created automatically by Jules for task [5458223176416215149](https://jules.google.com/task/5458223176416215149) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

このPRは `invites.ts` の `PUT/PATCH /api/invites/:id` エンドポイントに対する網羅的なテストを追加し、同時にDBエラー発生時のハンドリングを強化しています。

- **エラーハンドリング改善**: `invites.ts` で `enableForeignKeys` の呼び出しを `try` ブロック内に移動し、FK設定失敗時も適切に500を返すよう修正。
- **テスト追加**: 不正なアクション値(400)、招待不在(404)、権限外ユーザー(403)、既応答済み(400)、`db.batch`/`db.update` エラー伝播(500)など未カバーだったパスを全てテストで検証。
- **既存テスト改善**: 重複していた "invalid JSON body" テストを削除し、`expect(mockDb.select).not.toHaveBeenCalled()` アサーションを追加（前回レビューコメントへの対応）。
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

変更はテストコードと既存ハンドラーへのエラーハンドリング追加のみで、本番ロジックの動作変更はない。安全にマージ可能。

本番コードの変更は enableForeignKeys と DB操作を try/catch で囲む最小限の修正のみ。追加されたテストは実装と整合しており、モック構造も正しい。ロジックの変更は一切なく、エラーパスが適切に保護された。

特に注意が必要なファイルはない。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/api/src/routes/invites.ts | enableForeignKeys とDB操作を try/catch ブロックで囲み、失敗時に500を返すよう修正。ロジックは正しく、テストで動作が検証されている。 |
| apps/api/src/routes/__tests__/invites.test.ts | 新たに8つのエッジケーステストを追加し、100%カバレッジを達成。db.run のモックで enableForeignKeys の失敗をシミュレートしており、実装と一致している。 |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Client
    participant Handler as respondInviteHandler
    participant FK as enableForeignKeys
    participant DB as D1 Database

    Client->>Handler: PUT/PATCH /api/invites/:inviteId
    Handler->>Handler: JSON parse (400 if invalid)
    Handler->>Handler: action validation (400 if not accept/decline)
    Handler->>FK: enableForeignKeys(db)
    FK->>DB: db.run PRAGMA foreign_keys ON
    alt FK setup fails
        DB-->>Handler: 500 Failed to respond to invite
    end
    Handler->>DB: db.select invite by id
    alt DB select fails
        DB-->>Handler: 500 Failed to respond to invite
    end
    alt Invite not found
        DB-->>Handler: 404 Invite not found
    end
    alt Wrong user
        DB-->>Handler: 403 Forbidden
    end
    alt Already responded
        DB-->>Handler: 400 Invite already responded
    end
    alt "action = accept"
        Handler->>DB: db.batch update invite and insert paperAuthor
        alt batch fails
            DB-->>Handler: 500 Failed to respond to invite
        end
    else "action = decline"
        Handler->>DB: db.update coauthorInvites
        alt update fails
            DB-->>Handler: 500 Failed to respond to invite
        end
    end
    DB-->>Handler: ok
    Handler-->>Client: 200 ok true status accepted or declined
```

<sub>Reviews (7): Last reviewed commit: ["test: cover PUT decline invite response"](https://github.com/hiroki-org/openshelf/commit/a93d710f0db0f393cd2d3e2043d23b832ff0e7ff) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=32546927)</sub>

### CODE REVIEW

File: apps/api/src/routes/__tests__/invites.test.ts [hunk 1]
```diff
@@ -1,191 +1,421 @@
...
+      "http://localhost/api/invites/received",
+      {
+        headers: { Authorization: `Bearer ${token}` },
+      },
+      env as any,
+    );
+
+    expect(res.status).toBe(200);
+    const body = (await res.json()) as any;
+    expect(body.invites).toHaveLength(1);
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/invites.test.ts]: 対応しました: GET /api/invites/received のレスポンス全体を resolves.toEqual で検証するようにしました。検証: npm --workspace apps/api run test -- src/routes/__tests__/invites.test.ts, npm run lint, npm run typecheck.

File: apps/api/src/routes/__tests__/invites.test.ts [hunk 2]
```diff
@@ -1,191 +1,421 @@
...
-                },
-                body: "{ invalid }"
-            },
-            env as any
-        );
-
-        expect(res.status).toBe(400);
-        const body = await res.json() as any;
-        expect(body).toEqual({ error: "Invalid JSON body" });
+    expect(res.status).toBe(404);
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/invites.test.ts:334]: 対応しました: invite not found の404レスポンスbodyを { error: "Invite not found" } として検証しました。検証: npm --workspace apps/api run test -- src/routes/__tests__/invites.test.ts, npm run lint, npm run typecheck.

File: apps/api/src/routes/__tests__/invites.test.ts [hunk 3]
```diff
@@ -1,191 +1,421 @@
...
-                    Authorization: `Bearer ${token}`,
-                    "Content-Type": "application/json"
-                },
-                body: JSON.stringify({ inviteeId: "user-1" })
-            },
-            env as any
-        );
-
-        expect(res.status).toBe(400);
+    expect(res.status).toBe(403);
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/invites.test.ts:456]: 対応しました: Forbidden の403レスポンスbodyを { error: "Forbidden" } として検証しました。検証: npm --workspace apps/api run test -- src/routes/__tests__/invites.test.ts, npm run lint, npm run typecheck.

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
I have fixed the issue. Added body validation to the 403 test.

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
I have fixed the issue. Added body validation to the 403 test. Since it's a test addition and pre-commit checks pass, I think it's good to merge!

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
I've fixed it in another branch `test-coauthor-error-paths-5458223176416215149` because I couldn't push back to the original branch. However, I have added `expect(body).toEqual({ error: 'Forbidden' })` to the 403 test for invite body validation!

**Comment by is0692vs** [comment]
Oh, I see there were merge commits and changes made independently in branch! Thank you for the review, it seems I should not push, but I can see you merged the changes anyway!
